From 0388d79f8200705edc669065010a584b31d76678 Mon Sep 17 00:00:00 2001
From: Felix Radensky <felix.r@variscite.com>
Date: Tue, 3 Dec 2019 17:10:59 +0200
Subject: [PATCH] usb: ehci-mx6: Fix USB type detection

When external connector is defined the USB type validy check fails.
Fix the check to take external connector into account.

---
 drivers/usb/host/ehci-mx6.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/host/ehci-mx6.c b/drivers/usb/host/ehci-mx6.c
index 305c4819cc..c5973294ac 100644
--- a/drivers/usb/host/ehci-mx6.c
+++ b/drivers/usb/host/ehci-mx6.c
@@ -310,7 +310,7 @@ static int ehci_usb_ofdata_to_platdata(struct udevice *dev)
 	struct usb_platdata *plat = dev_get_platdata(dev);
 	struct ehci_mx6_priv_data *priv = dev_get_priv(dev);
 	enum usb_dr_mode dr_mode;
-	const struct fdt_property *extcon;
+	const struct fdt_property *extcon = NULL;
 
 	extcon = fdt_get_property(gd->fdt_blob, dev_of_offset(dev),
 			"extcon", NULL);
@@ -335,7 +335,7 @@ static int ehci_usb_ofdata_to_platdata(struct udevice *dev)
 	};
 
 check_type:
-	if (priv->init_type != USB_INIT_UNKNOWN && priv->init_type != plat->init_type) {
+	if (priv->init_type != USB_INIT_UNKNOWN && !extcon && priv->init_type != plat->init_type) {
 		debug("Request USB type is %u, board forced type is %u\n",
 			plat->init_type, priv->init_type);
 		return -ENODEV;
